src/ipc_protocol/protocol.rs

```diff
@@ -367,6 +367,19 @@ impl ProtocolClient {
         }
     }
 
+    pub async fn circular_arc(&self, id: TurtleId, radius: Distance, extent: Radians, direction: RotationDirection) {
+        if radius.is_normal() && extent.is_normal() {
+            let steps = 250;    // Arbitrary value for now.
+            let step = radius.abs() * extent.to_radians() / steps as f64;
+            let rotation = radius.signum() * extent / steps as f64;
+
+            for _ in 0..steps {
+                self.move_forward(id, step).await;
+                self.rotate_in_place(id, rotation, direction).await;
+            }
+        }
```

Comment:
Actually, I think when there is only one condition to check, it is easier to understand in the first form, because it lets us think in the positive way; whereas when having multiple conditions in separate regions, the second form is better in order to avoid a lot of combinations and nesting. So I'd like to keep it as is, if that's okay with you.

Reply:
Let's make the change to keep this code consistent with the rest of the file. I don't think this condition is important enough to wrap the rest of the code in the function. It's fairly easy to understand that we're checking `is_normal` and returning early if either parameter is a value we can't use.